src/cargo/util/toml/mod.rs

```diff
@@ -773,6 +1071,179 @@ impl<'de> de::Deserialize<'de> for VecStringOrBool {
     }
 }
 
+#[derive(Serialize, Clone, Debug)]
+#[serde(untagged)]
+pub enum MaybeWorkspace<T> {
+    Workspace,
```

Comment:
Could this leverage serde instead of writing a custom `Deserialize` by defining this as `Workspace { workspace: bool }`?

Reply:
Or something like `Workspace(TomlWorkspaceField)`

Reply:
FWIW with a custom `Deserialize` implementation this should be able to customize error messages as well, I think we have one or two implementations that do that already?